diff --git a/examples/mpi/CMakeLists.txt b/examples/mpi/CMakeLists.txt
index b42a251..ba85c6c 100644
--- a/examples/mpi/CMakeLists.txt
+++ b/examples/mpi/CMakeLists.txt
@@ -1,3 +1,9 @@
+find_package(ExaTN QUIET)
+if (ExaTN_FOUND)
+   message(STATUS "Found ExaTN at ${EXATN_ROOT}")
+   include_directories(${EXATN_INCLUDE_DIRS})
+endif()
+
 include_directories(${XACC_INCLUDE_DIRS})
 include_directories(${PROJECT_SOURCE_DIR})
 
@@ -8,4 +14,7 @@
 target_link_libraries(qaoa_mpi PRIVATE xacc::xacc)
 
 add_executable(deutsch_jozsa deutsch_jozsa.cpp)
-target_link_libraries(deutsch_jozsa PRIVATE xacc::xacc)
\ No newline at end of file
+target_link_libraries(deutsch_jozsa PRIVATE xacc::xacc)
+
+add_executable(circuit_runner circuit_runner.cpp)
+target_link_libraries(circuit_runner PRIVATE xacc::xacc exatn::exatn)
diff --git a/tnqvm/TNQVM.hpp b/tnqvm/TNQVM.hpp
index 047c4d0..0a213a7 100644
--- a/tnqvm/TNQVM.hpp
+++ b/tnqvm/TNQVM.hpp
@@ -135,7 +135,7 @@ public:
   virtual void
   execute(std::shared_ptr<AcceleratorBuffer> buffer,
           const std::shared_ptr<CompositeInstruction> baseCircuit,
-          const std::vector<std::shared_ptr<CompositeInstruction>> basisRotations) override;
+          const std::vector<std::shared_ptr<CompositeInstruction>> basisRotations);
 
   const std::vector<std::complex<double>>
   getAcceleratorState(std::shared_ptr<CompositeInstruction> program) override;
